src/Bang.Analyzers.Tests/BangAnalyzerVerifier.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections.Immutable;
+using Bang.Components;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Testing.Verifiers;
+using Microsoft.CodeAnalysis.CSharp.Testing;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Testing;
+
+namespace Bang.Analyzers.Tests;
+
+/// <summary>
+/// Verifier that includes Bang binaries to the project.
+/// </summary>
+/// <typeparam name="TAnalyzer">Analyzer under test.</typeparam>
+public class BangAnalyzerVerifier<TAnalyzer> : AnalyzerVerifier<TAnalyzer, BangTest<TAnalyzer>, XUnitVerifier>
+	where TAnalyzer : DiagnosticAnalyzer, new() { }
+	
+/// <summary>
+/// Implementation of CSharpAnalyzerTest that uses net7.0 (not enabled by default
+/// as of now and needed for Bang) and includes the Bang dlls.
+/// </summary>
+/// <typeparam name="TAnalyzer">Analyzer under test.</typeparam>
+public sealed class BangTest<TAnalyzer> : CSharpAnalyzerTest<TAnalyzer, XUnitVerifier>
+	where TAnalyzer : DiagnosticAnalyzer, new()
+{
+	public BangTest()
+	{
+		var bangReference = MetadataReference.CreateFromFile(typeof(IComponent).Assembly.Location);
+		TestState.AdditionalReferences.Add(bangReference);
+		ReferenceAssemblies = Net.Net70;
+	}
+}
+
+/// <summary>
+/// This is kind of a hack because, as of now, net7.0 is not available out-of-the-box for analyzer testing.
```

Comment:
Are they adding support in net8.0? I wonder if we should just switch .net 8 pretty soon...

Reply:
I'm not sure about the roadmap, but adapting this Hack to work with net8.0 should be easy enough